python/pyspark/sql/functions.py

```diff
@@ -3765,12 +3765,12 @@ def collect_set(col: "ColumnOrName") -> Column:
     Example 1: Collect values from a single column DataFrame
 
     >>> from pyspark.sql import functions as sf
-    >>> df = spark.createDataFrame([(2,), (5,), (5,)], ('age',))
+    >>> df = spark.createDataFrame([(5,), (5,), (5,)], ('age',))
     >>> df.select(sf.collect_set('age')).show()
     +----------------+
     |collect_set(age)|
     +----------------+
-    |          [5, 2]|
+    |             [5]|
```

Comment:
Sorry, this doctest produced different results when using Scala 2.12 and Scala 2.13.

When using Scala 2.12, this test case prints `[5, 2]`, but when using Scala 2.13, this test case prints `[2, 5]`. Skip or simplify it? Which one do you think is better? @zhengruifeng @HyukjinKwon 

https://github.com/apache/spark/actions/runs/6209111340/job/16856005714

```
**********************************************************************
File "/__w/spark/spark/python/pyspark/sql/connect/functions.py", line 1030, in pyspark.sql.connect.functions.collect_set
Failed example:
    df.select(sf.collect_set('age')).show()
Expected:
    +----------------+
    |collect_set(age)|
    +----------------+
    |          [5, 2]|
    +----------------+
Got:
    +----------------+
    |collect_set(age)|
    +----------------+
    |          [2, 5]|
    +----------------+
    <BLANKLINE>
**********************************************************************
   1 of   9 in pyspark.sql.connect.functions.collect_set
***Test Failed*** 1 failures.

Had test failures in pyspark.sql.connect.functions with python3.9; see logs.
Error:  running /__w/spark/spark/python/run-tests --modules=pyspark-connect --parallelism=1 ; received return code 255
Error: Process completed with exit code 19.
```

Reply:
I think it can be expected, since `collect_set/collect_list` do not guarantee the ordering, what about adding a `array_sort` to make it deterministic?

Reply:
make sense